### PR TITLE
[RI2 Ch01] Brief description of deployment scenarios

### LINKED ISSUE #1791: [RI-2 Ch1] Add section about host provisioning
Add a high-level description of the role of host provisioning  in RI-2: bare-metal provisioning, bare-metal cloud infra provider, IaaS provider.

### PR DESCRIPTION
**Pull request by gkunz**
This patch adds a brief overview of the deployment scenarios of RI2
and their impact on host provisioning.

Fixes #1791

Signed-off-by: Georg Kunz <georg.kunz@est.tech>

### CODE REVIEW

File: doc/ref_impl/cntt-ri2/chapters/chapter01.md [hunk 1]
```diff
@@ -91,7 +91,19 @@ The following subsections provide an overview of the provisioning stages as they
...
-> Add high level description of host provisioning stage
+The RI should be deployable on different infrastructures. Specifically, the following deployment scenarios are in scope of the RI:
+
+- **Deployment on bare-metal hosts**
+   This deployment scenario is typically used in internal labs and existing OPNFV community labs. It requires the RI installation tooling to provision empty bare-metal machines with a base operating system and to apply a baseline network configuration.
+
+- **Deployment on bare-metal hosts provided by a cloud provider**
+   This deployment scenario makes use of bare-metal cloud providers such as Packet.net. The machines provided in this case have been pre-provisioned with a base operating system and optionally a corresponding network configuration. This scenario requires the RI installation tooling to request and configure compute resources via the API of the cloud provider.
+
+- **Deployment on an Infrastructure-as-aa-Service Layer**
```
Comment (tomkivlin) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.md:102]:
```suggestion
- **Deployment on an Infrastructure-as-a-Service Layer**
```

**Review by tomkivlin — CHANGES_REQUESTED** [review]

**Comment by tomkivlin** [comment]
As well as the three listed, should we include the ability for "any" infra to be used (i.e. if someone wants to provide their own) but then use the RI2 tooling to deploy Kubernetes + add-ons to that infrastructure.  So RI2 tooling wouldn't deploy the infra (which the current three options assume), but does deploy the k8s layer.

**Comment by gkunz** [comment]
> As well as the three listed, should we include the ability for "any" infra to be used (i.e. if someone wants to provide their own) but then use the RI2 tooling to deploy Kubernetes + add-ons to that infrastructure. So RI2 tooling wouldn't deploy the infra (which the current three options assume), but does deploy the k8s layer.

Good point. Implicitly this is included in the general approach of having a host provision phase and a k8s deployment phase. So, the second phase should be runnable without the first one which is inline with my perspective. I'll check how to make this more clear in the text.

**Review by tomkivlin — APPROVED** [review]

File: doc/ref_impl/cntt-ri2/chapters/chapter01.md [hunk 2]
```diff
@@ -85,12 +85,12 @@ The approach this RI will take is to separate out the deployment in to two layer
...
+Meaning, initially, the RI is not looking to have a single installer that can both build out virtual or physical machines **and** build out the Kubernetes and other components. The primary reason for this was to ensure the loose coupling between the two layers, to drive the concept that RI2 is a standalone platform that can (in theory at least) be deployable to any infrastructure, whether that be some physical machines in a lab, or virtual machines in a private or public cloud environment, for example. From am implementation perspective, this means that the first step - infrastructure provisioning - needs to potentially support many different instructures and could even be an optional step in the overall end-to-end deployment process if an infrastructure is provided by other means. To tightly intertwine the deployment of machines and the Kubernetes platform with a single installer would potentially limit the locations on which the RI can be installed.
```
Comment (rihabbanday) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.md:88]:
```suggestion
Meaning, initially, the RI is not looking to have a single installer that can both build out virtual or physical machines **and** build out the Kubernetes and other components. The primary reason for this was to ensure the loose coupling between the two layers, to drive the concept that RI2 is a standalone platform that can (in theory at least) be deployable to any infrastructure, whether that be some physical machines in a lab, or virtual machines in a private or public cloud environment, for example. From am implementation perspective, this means that the first step - infrastructure provisioning - needs to potentially support many different infrastructures and could even be an optional step in the overall end-to-end deployment process if an infrastructure is provided by other means. To tightly intertwine the deployment of machines and the Kubernetes platform with a single installer would potentially limit the locations on which the RI can be installed.
```

**Review by michaelspedersen — APPROVED** [review]
lgtm

**Comment by tomkivlin** [comment]
@rihabbanday if you're happy with the changes made, could you approve please?  Should be able to merge then.

**Review by rihabbanday — APPROVED** [review]

**Pull request merged**